editoast/src/schema/rolling_stock/light_rolling_stock.rs

```diff
@@ -17,66 +18,67 @@ crate::schemas! {
     LightEffortCurves,
 }
 
-#[derive(Debug, QueryableByName, Deserialize, Serialize, ToSchema)]
+#[derive(Debug, Deserialize, Serialize, ToSchema)]
 pub struct LightRollingStock {
-    #[diesel(sql_type = BigInt)]
     pub id: i64,
-    #[diesel(sql_type = Text)]
     pub name: String,
-    #[diesel(sql_type = Text)]
     pub railjson_version: String,
-    #[diesel(sql_type = Bool)]
     pub locked: bool,
-    #[diesel(sql_type = Jsonb)]
-    #[schema(value_type = LightEffortCurves)]
-    pub effort_curves: DieselJson<LightEffortCurves>,
-    #[diesel(sql_type = Nullable<Text>)]
+    pub effort_curves: LightEffortCurves,
     pub base_power_class: Option<String>,
-    #[diesel(sql_type = Double)]
     pub length: f64,
-    #[diesel(sql_type = Double)]
     pub max_speed: f64,
-    #[diesel(sql_type = Double)]
     pub startup_time: f64,
-    #[diesel(sql_type = Double)]
     pub startup_acceleration: f64,
-    #[diesel(sql_type = Double)]
     pub comfort_acceleration: f64,
-    #[diesel(sql_type = Jsonb)]
-    #[schema(value_type = Gamma)]
-    pub gamma: DieselJson<Gamma>,
-    #[diesel(sql_type = Double)]
+    pub gamma: Gamma,
     pub inertia_coefficient: f64,
-    #[diesel(sql_type = Double)]
     pub mass: f64,
-    #[diesel(sql_type = Jsonb)]
-    #[schema(value_type = RollingResistance)]
-    pub rolling_resistance: DieselJson<RollingResistance>,
-    #[diesel(sql_type = Text)]
+    pub rolling_resistance: RollingResistance,
     #[schema(value_type = LoadingGaugeType)]
     pub loading_gauge: String,
-    #[diesel(sql_type = Jsonb)]
-    #[schema(value_type = RollingStockMetadata)]
-    pub metadata: DieselJson<RollingStockMetadata>,
-    #[diesel(sql_type = Jsonb)]
-    #[schema(value_type = HashMap<String, String>)]
-    pub power_restrictions: DieselJson<HashMap<String, String>>,
-    #[diesel(sql_type = Jsonb)]
-    #[schema(value_type = Vec<EnergySource>)]
-    pub energy_sources: DieselJson<Vec<EnergySource>>,
+    pub metadata: RollingStockMetadata,
+    #[schema(required)]
+    pub power_restrictions: HashMap<String, String>,
+    pub energy_sources: Vec<EnergySource>,
     #[serde(skip)]
-    #[diesel(sql_type = BigInt)]
     pub version: i64,
-    #[diesel(sql_type = Array<Text>)]
-    pub supported_signaling_systems: Vec<String>,
+    pub supported_signaling_systems: RollingStockSupportedSignalingSystems,
 }
 
-#[derive(Debug, QueryableByName, Deserialize)]
+impl From<LightRollingStockModel> for LightRollingStock {
```

Comment:
We should move this to the `craste::modelv2::light_rolling_stock` module. The idea is to avoid dependencies from models to schema. Check "split editoast in crates" issue.

Reply:
Done.